design/one-pager-merge-jets-proposal.md

```diff
@@ -0,0 +1,370 @@
+Crossplane providers. It seems like Terraform has achieved this goal only for
+GCP; Azure and AWS Terraform providers are not maintained by the vendors. In our
+case, Crossplane is a CNCF project with an open governance model, so we can't
+really compare it with Terraform, a property of HashiCorp, in that respect but
+it still provides a data point.
+
+The strategy we choose should take the following possibilities into account:
+* They may choose to bootstrap their own Crossplane provider, maybe even not
+under Crossplane organization.
+* They might decide to convert the underlying implementation to their own
+technology, similar to GCP Terraform provider that is [undergoing a
+process][gcp-tf-contributing] where they change implementation from handwritten
+API calls to Magic Modules and DCL.
+* They would provide the tools but not take part in maintenance of the codebase.
+
+In the first case, regardless of whether we go with separate or mixed providers,
+we will likely suggest users to migrate to the vendor-maintained provider so
+that they can get commercial grade support for the bugs in the provider. One
+could argue that if users are on a single provider once that happens, it's
+probably easier to handle the migration. In addition, it's likely the main
+reason they'd step in would be community-maintained provider having a large user base,
+so it's not very likely that they will start a new provider and handle migration
+from a provider with such a large user base.
+
+In the second case, mixed provider wouldn't require any manual migration. The
+conversion webhooks that Kubernetes API provides will take care of the API
+changes as described in [provider strategy doc][strategy-doc] in detail. In the
+case of separate providers though, users of one of the provider would have to
+migrate.
+
+In the third case, if we go with single mixed provider, we can choose to use the
+cloud vendor tooling for the resources we decide to convert. If we go with
+separate providers, we could keep working with those tools, like AWS CC, in the
+existing classic provider or bootstrap a new one but either case would either be
+similar or worse than the single mixed provider in terms of choices users would
+have to consider. Overall, the decision of how to go from Terrajet to
+cloud-vendor tooling is an independent decision from how we manage Terrajet and
+classic provider mix since it's the step after we decide that, if ever, the
+provided tooling is better than we have. For example, today, calling AWS API
+directly is definitely not better than Terrajet.
+
+In all cases, our best bet on getting them to maintain their provider is to
+increase usage so that their customers demand it and the single mixed provider
+makes people more comfortable by reducing the decision overhead and providing
+confidence to users about what they can expect to happen to the tool they build
+their infrastructure on top of.
+
+### Convergence of Efforts
```

Comment:
Honestly I don't think mixing providers alone makes a big (or really any) difference here. Contributors still have to account for N different implementations (TJ, classic, etc) when adding a feature or functionality that touches many MRs, those N different implementations just all live in one repo instead of several.

Reply:
My point about in this section is actually more like usage divergence which would result contribution divergence. For example, contributor A fixes an EKS bug in Jet provider but classic provider users don't get that fix until someone fixes it there, too. Or that bug doesn't exist in classic at all, so if contributor A were using classic, then they'd not have to spend that effort in the first place. It's essentially about the upsides of having a common single controller code for a single API resource.

> Contributors still have to account for N different implementations (TJ, classic, etc) when adding a feature or functionality that touches many MRs, those N different implementations just all live in one repo instead of several.

That is true but I think that makes up a really low amount of the contributions to providers and we had tried to make Jet ones as similar as possible to the classic ones to reduce the cost of difference. In my mind, I think the API-specific features/fixes are more critical to think about in that respect.